In [23]:
import pandas as pd
from statistics import mode

In [2]:
db = pd.read_csv("2_titanic_dataset.csv")

In [4]:
db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


In [5]:
db.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

###  1.How many people travel in the Titanic?

In [11]:
db.shape[0]

891

##### Knowledge retrieved: There were 891 people in the Titanic

### 2. Does someone has the excact same age?

In [34]:
mode(db["Age"].unique())

StatisticsError: no unique mode; found 89 equally common values

##### Knowledge retrieved: There are no persons with the excact same age.

###  3.How many values are we missing for each column?

In [38]:
db.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

##### Knowledge retrieved: We are missing 177 ages and 687 cabin description

###  4.How many people of each class has travelled?

In [52]:
upper = sum(db['Pclass'] == 1)
middle = sum(db['Pclass'] == 2)
lower = sum(db['Pclass'] == 3)

there were 216 upper class in Titanic
there were 184 upper class in Titanic
there were 491 upper class in Titanic


##### Knowledge retrieved:
###### >there were 216 upper class in Titanic
###### >there were 184 middle class in Titanic
###### >there were 491 lower class in Titanic

###  5.How many people has survived and how many died?

In [76]:
survived = sum(db['Survived'] == 1)
died = sum(db['Survived'] == 0)

##### Knowledge retrieved: 342 people survied adn 549 died.

In [102]:
###  7.What was the average of the Fare price?

In [103]:
db["Fare"].mean()

32.2042079685746

###  7.How many people of each class has died?

In [96]:
upperdied = sum(db['Pclass'] == 1 if (db['Survived'] == 0)
middledied = sum(db['Pclass'] == 2) and sum(db['Survived'] == 0)
lowerdied = sum(db['Pclass'] == 3) and sum(db['Survived'] == 0)
print(upperdied)
print(middledied)
print(lowerdied)

SyntaxError: invalid syntax (<ipython-input-96-03a91394f67f>, line 2)

In [111]:
upperdied = db.loc[(db['Pclass'] == 1) & (db['Survived'] == 0)].sum()
upperdied

PassengerId                                                32824
Survived                                                       0
Pclass                                                        80
Name           McCarthy, Mr. Timothy JFortune, Mr. Charles Al...
Sex            malemalemalemalemalemalemalemalemalemalemalema...
Age                                                       2796.5
SibSp                                                         23
Parch                                                         24
Ticket         1746319950PC 17601PC 1760411378911350936973PC ...
Fare                                                     5174.72
Embarked       SSCCSCSCSSCSSCSSCCSSCCSQSSSSCSCSSSSSSCCSSCSSSS...
dtype: object

In [60]:
##### Knowledge retrieved:
###### >xxx upper class people in Titanic
###### >xxx middle class people in Titanic
###### >xxx lower class people in Titanic